developer-hub/src/reference_docs/defensive_programming.rs

```diff
@@ -0,0 +1,558 @@
+//! As our runtime should _never_ panic; this includes carefully handling [`Result`]/[`Option`]
+//! types, eliminating the possibility of integer overflows, converting between number types, or
+//! even handling floating point usage with fixed point arithmetic to mitigate issues that come with
+//! floating point calculations.
+//!
+//! Intentional and predictable design should be our first and foremost
+//! property for ensuring a well running, safely designed system.
+//!
+//! ## Defensive Programming
+//!
+//! Defensive programming is a design paradigm that enables a particular program to continue
+//! running despite unexpected behavior. These unforseen circumstances may
+//! cause the program to stop, (or in the Rust context, `panic!`) defensive practices allow for
+//! these circumstances to be accounted for ahead of time, and for them to be handled in a more
+//! graceful manner.
+//!
+//! The Polkadot SDK is both built to reflect these principles, and to be used accordingly.
+//!
+//! ## General Practices
+//!
+//! When developing within the context of the a runtime, there is one golden rule:
```

Comment:
Since this ref doc is describing safe programming in Substrate in general, I think it would be interesting to elaborate *why* the runtime should not panic (no fee DoS and, in some cases, stalled runtime) and where it's OK.

Reply:
Added one example, it may need more elaboration.